src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java

```diff
@@ -60,21 +63,47 @@ public final String getCredentialsId() {
         return credentialsId;
     }
 
-    /** Callback for processing during a build. */
-    public interface MultiEnvironment {
+    /** Result of {@link #bind}. */
+    public static final class MultiEnvironment implements Serializable {
 
-        /** Produces the value of the environment variables. */
-        Map<String,String> values();
+        private final Map<String,String> values;
+        private final Unbinder unbinder;
 
+        public MultiEnvironment(Map<String,String> values) {
+            this(values, new NullUnbinder());
+        }
+
+        public MultiEnvironment(Map<String,String> values, Unbinder unbinder) {
+            this.values = new HashMap<String,String>(values);
+            this.unbinder = unbinder;
+        }
+
+        public Map<String,String> getValues() {
+            return Collections.unmodifiableMap(values);
+        }
+
+        public Unbinder getUnbinder() {
+            return unbinder;
+        }
+
+    }
+
+    /** Callback run at the end of a build. */
+    public interface Unbinder extends Serializable {
         /** Performs any needed cleanup. */
-        void unbind() throws IOException, InterruptedException;
+        void unbind(@Nonnull Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException;
+    }
```

Comment:
I suppose this is an internal interface of which there are no external implementations?